.gitlab/functional_test/regression_detector.yml

```diff
@@ -180,7 +214,18 @@ single-machine-performance-regression_detector:
       else:
           print("Quality gate passed.")
           sys.exit(0)
-      EOF
+      EOF 
+    # Check the exit code and handle failures
+    - |
+      exit_code=$?
```

Comment:
Part of me wants to leverage the machine-readable json report to report more specific attributes, ie "Failed due to an optimization goal comparison" and "failed due to a bounds check failure" and "failed quality gate assertions".

However that does increase the reliance on the specific formatting of the machine readable report. 
@GeorgeHahn what do you think we should do here?

Reply:
I'm a bit surprised not to see `smp job result` show up in this workflow. That would give us pass/fail on optimization goals and on bounds checks, and then the check here would track quality gates failures.

I suppose they're not currently necessary because they don't fail the job. I think those cases would be helpful to track. They may not block jobs, but they are both user facing signals that we can monitor for anomalies.

The specific invocations to check would be [this](https://github.com/DataDog/datadog-agent/pull/31017/files#diff-d9ba3a3d702a285359c8bde5b64ff13e38de06029bc8ed0d4ce775eb24178dedR336-R338) and the same with `--signal regression-detector`.

Reply:
I was surprised to not see `job result` also. I can add that back and catch the exit 1 and just tag accordingly without exiting the script. Currently having some troubles on how to catch errors (and thus tagging the error) from the embedded python but working on that today

Reply:
The Python script emulates `smp job result` and also does filtering by experiment name that would be difficult to do with the current `smp job result` implementation; IIRC, `smp job result` reads a JSON file with the pass or fail status of the entire job. There's no reason we couldn't change that later -- I think the Python script was written to be expedient.